language/tools/vm-genesis/src/lib.rs

```diff
@@ -385,15 +385,25 @@ pub fn validator_registrations(node_configs: &[NodeConfig]) -> Vec<ValidatorRegi
             let network = n.validator_network.as_ref().unwrap();
             let network_keypairs = network.network_keypairs.as_ref().unwrap();
             let signing_key = network_keypairs.signing_keypair.public_key();
-            let raw_advertised_address =
-                RawNetworkAddress::try_from(&network.advertised_address).unwrap();
+
+            let identity_key = network_keypairs.identity_keypair.public_key();
+            let handshake_version = network.handshake_version;
+
+            let advertised_address = network
+                .advertised_address
+                .clone()
+                .into_prod(identity_key, handshake_version);
+            let raw_advertised_address = RawNetworkAddress::try_from(&advertised_address).unwrap();
```

Comment:
i'm not sure i understand this comment? `RawNetworkAddress` is tackling a totally orthogonal issue:

```rust
/// A `RawNetworkAddress` is the serialized, unverified, on-chain representation
/// of a [`NetworkAddress`]. Specifically, a `RawNetworkAddress` is usually an
/// [`lcs`]-serialized `NetworkAddress`.
///
/// This representation is useful because:
///
/// 1. Move does't understand (and doesn't really need to understand) what a
///    `NetworkAddress` is, so we can just store an opaque `vector<u8>` on-chain,
///    which we represent as `RawNetworkAddress` in Rust.
/// 2. We want to deserialize a `Vec<NetworkAddress>` but ignore ones that don't
///    properly deserialize. For example, a validator might advertise several
///    `NetworkAddress`. If that validator is running a newer version of the node
///    software, and old versions can't understand one of the `NetworkAddress`,
///    they would be able to ignore the one that doesn't deserialize for them.
///    We can easily do this by storing a `vector<vector<u8>>` on-chain, which
///    we deserialize as `Vec<RawNetworkAddress>` in Rust. Then we deserialize
///    each `RawNetworkAddress` into a `NetworkAddress` individually.
///
/// Note: deserializing a `RawNetworkAddress` does no validation, other than
/// deserializing the underlying `Vec<u8>`.
```

i also don't understand the comment about not "leveraging the non-human-readable ser/de". all network addresses stored on-chain and sent over-the-wire (the important places) are in the compact binary format. for our prod addresses, the compact format is about 35-40% the size of the human readable format

```rust
let addr_str = "/ip4/123.123.123.123/tcp/55555/ln-noise-ik/080e287879c918794170e258bfaddd75acac5b3e350419044655e4983a487120/ln-handshake/233";
let addr = NetworkAddress::from_str(addr_str).unwrap();
let raw_addr = RawNetworkAddress:try_from(&addr).unwrap();

addr_str.len()
// => 124 bytes

raw_addr.len()
// => 45 bytes

45f / 124
// => 0.362..
```

Reply:
Yes, this is totally orthogonal to the PR, but I feel like we should explore merging RawNetworkAddress concepts directly into NetworkAddress, it isn't clear to me why we want both? For example in other code, you convert from NetworkAddress to RawNetworkAddress using a try_from... but that should be infallible afaiu. if it were all self-contained this would be more self-evident.